function-processor/src/main/java/com/dylibso/chicory/function/processor/FunctionProcessor.java

```diff
@@ -86,8 +96,13 @@ private void processModule(TypeElement type) {
                 functions.add(processMethod(member, (ExecutableElement) member, moduleName));
             }
         }
-
-        var cu = new CompilationUnit(type.getEnclosingElement().toString());
+        var pkg = getPackageName(type);
+        var packageName = pkg.getQualifiedName().toString();
+        var cu = (pkg.isUnnamed()) ? new CompilationUnit() : new CompilationUnit(packageName);
+        if (!pkg.isUnnamed()) {
+            cu.setPackageDeclaration(packageName);
+            cu.addImport(type.getQualifiedName().toString());
```

Comment:
Is this needed? Imports aren't needed for the same package.

Also, I don't see any changes in `BasicMathGenerated` or `SimpleGenerated`.

Reply:
It's part of the fix:
https://github.com/dylibso/chicory/pull/510/files#diff-bd5d5e9bbce4b5d75cb3f17f85ffc906bfdb7125dfa7e37722381bcbfd701f05R3

the import is added if it differs from the `packageName` (e.g. when using nested classes).